src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateContestedSubmittedHandlerTest.java

```diff
@@ -38,38 +42,72 @@ public class SolicitorCreateContestedSubmittedHandlerTest {
+    void givenACcdCallbackSolicitorCreateContestedCase_WhenCanHandleCalled_thenHandlerCanHandle() {
         assertThat(handler
                 .canHandle(CallbackType.SUBMITTED, CaseType.CONTESTED, EventType.SOLICITOR_CREATE),
             is(true));
     }
 
     @Test
-    public void givenACcdCallbackAboutToSubmit_WhenCanHandleCalled_thenHandlerCanNotHandle() {
+    void givenACcdCallbackAboutToSubmit_WhenCanHandleCalled_thenHandlerCanNotHandle() {
         assertThat(handler
                 .canHandle(CallbackType.ABOUT_TO_SUBMIT, CaseType.CONTESTED, EventType.SOLICITOR_CREATE),
             is(false));
     }
 
     @Test
-    public void givenACcdCallbackSolicitorCreateContestedCase_WhenHandle_thenAddSupplementary() {
-        FinremCallbackRequest callbackRequest = buildFinremCallbackRequest();
-        handler.handle(callbackRequest, AUTH_TOKEN);
+    void givenACcdCallbackSolicitorCreateContestedCase_WhenHandleWithAppRepresentation_thenAddSupplementaryAndAppRep() {
+        FinremCallbackRequest callbackRequest = buildFinremCallbackRequest(YesOrNo.YES);
+        GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> response = handler.handle(callbackRequest, AUTH_TOKEN);
 
-        verify(createCaseService, times(1)).setSupplementaryData(eq(callbackRequest), any());
+        verify(createCaseService, times(1)).setSupplementaryData(callbackRequest, AUTH_TOKEN);
+        verify(assignApplicantSolicitorService, times(1))
+            .setApplicantSolicitor(callbackRequest.getCaseDetails(), AUTH_TOKEN);
+        assertTrue(response.getErrors().isEmpty());
     }
 
-    private FinremCallbackRequest buildFinremCallbackRequest() {
-        FinremCaseData caseData = FinremCaseData.builder().build();
+    @Test
+    void givenACcdCallbackSolicitorCreateContestedCase_WhenHandleWithAppRepresentationError_thenAddSupplementaryAndHandleException() {
+        FinremCallbackRequest callbackRequest = buildFinremCallbackRequest(YesOrNo.YES);
+
+        doThrow(new AssignCaseAccessException("Exception message"))
+            .when(assignApplicantSolicitorService)
+            .setApplicantSolicitor(any(), any());
+
+        GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> response = handler.handle(callbackRequest, AUTH_TOKEN);
+
+        verify(createCaseService, times(1)).setSupplementaryData(callbackRequest, AUTH_TOKEN);
+        verify(assignApplicantSolicitorService, times(1))
+            .setApplicantSolicitor(callbackRequest.getCaseDetails(), AUTH_TOKEN);
+
+        assertTrue(response.getErrors().contains("Exception message"));
+        assertTrue(response.getErrors()
+            .contains("Failed to assign applicant solicitor to case, "
+                + "please ensure you have selected the correct applicant organisation on case"));
+    }
+
+    @Test
+    void givenACcdCallbackSolicitorCreateContestedCase_WhenHandleWithNoAppRepresentation_thenApplicantSolicitorNotAssigned() {
```

Comment:
I like the namings on these tests, nice 👍

Reply:
Thanks :)